core/src/main/java/com/datastax/oss/driver/api/core/config/DriverConfigProfile.java

```diff
@@ -67,6 +83,15 @@
 
   DriverConfigProfile withDuration(DriverOption option, Duration value);
 
+  /**
+   * Returns a representation of all the child options under a given option.
+   *
+   * <p>This is only used to compare configuration sections across profiles, so the actual
+   * implementation does not matter, as long as identical sections (same options with same values,
+   * regardless of order) compare as equal and have the same {@code hashCode()}.
+   */
+  Object getComparisonKey(DriverOption option);
```

Comment:
Not super happy about this. But we need a way to know when two profiles have the same configuration, in order to create a single policy instance that will serve them both. This also applies to the default profile and its children, because we have no way of knowing which options were inherited or redeclared.